lightdash/lightdash
# chore: vitest react testing library

related to: #8739 

### Description:

removes jest and adds vitest (frontend only!)

motivation:
- vitest is faster for compiling and transforming code (its based on vite)
  - see screenshots: from ~22s => ~2s
- use the same stack for transforming modules (build, dev and test is now handled with vite)
- vitest has compliant api to jest
- vitest configuration is simpler

- jest ![CleanShot 2024-01-26 at 16 43 21@2x](https://github.com/lightdash/lightdash/assets/962095/3736053c-6123-4d9d-9d84-3cf56ff27e45)
- vitest ![CleanShot 2024-01-26 at 17 02 33@2x](https://github.com/lightdash/lightdash/assets/962095/cd401e8b-9d13-43de-abf0-1c8343118794)

### Reviewer actions
